lib/index.js

```diff
@@ -0,0 +1,172 @@
+      name: 'SDK language variant',
+      id: 'variant',
+      availableOptions: getLanguageList(),
+      type: 'String',
+      description: 'Specifies Language variant for SDK generation'
+    },
+    includeReadme: {
+      name: 'Includes README with the package if output type is file',
+      id: 'includeReadme',
+      type: Boolean,
+      default: false,
+      description: 'Specifies to add or not to add README along with sdk'
+    }
+  };
+}
+
+/**
+ * Checks and sanitizes optoins provided to sdkgenerator
+
+ * @param {Object} sdkOptions - codegenoptions provided to generate method
+ * @param {Object} codegenOptions - sdkoptions provided to generate method
+ * @param {Function} callback - Callback function to return results (err, sdkgenoptions, codegenoptions);
+ */
+function checkOptions (sdkOptions, codegenOptions, callback) {
+  // checks for sdkgenerator
+  if (!_.isObject(sdkOptions)) {
+    return callback(errors.INVALID_PARAMETER, null, null);
+  }
+  if (sdkOptions.language === null) {
+    return callback(errors.INVALID_LANGUAGE, null, null);
+  }
+  if (sdkOptions.variant === null) {
+    sdkOptions.variant = sdkOptions.language;
+  }
+  if (sdkOptions.outputType === 'File' && !sdkOptions.outputFilePath) {
+    return callback(errors.FILE_PATH_NOT_PROVIDED, null, null);
+  }
+  // checks for codegenerator
+  if (!_.isObject(codegenOptions) || codegenOptions === null) {
+    return callback(errors.INVALID_PARAMETER, null, null);
+  }
+  if (_.isEmpty(codegenOptions)) {
+    codegenOptions.SDKGEN_enabled = true;
+  }
+  return callback(null, sdkOptions, codegenOptions);
+}
+
+/**
+ * Root Function which generates collection level code generator / Collection SDK
+
+ * @param {String} source - Source of the collection
+ * @param {Object} sdkOptions - Options for SDK generator
+ * @param {String} sdkOptions.language - Specifies language for SDK generator
+ * @param {String} sdkOptions.variant - Specifies language variant for SDK generator
+ * @param {String} [sdkOptions.outputType] - Output method ie; string/file
+ * @param {String} [sdkOptions.outputFilePath] - Output File path for sdk output
+ * @param {String} [sdkOptions.includeReadme] - Includes README along with the output.
+ * @param {Object} [codeOptions] - Options for code generator
+ * @param {Number} [codeOptions.indentType] - indentation based on Tab or spaces
+ * @param {Number} [codeOptions.indentCount] - count/frequency of indentType
+ * @param {Number} [codeOptions.requestTimeout] : time in milli-seconds after which request will bail out
+ * @param {Boolean} [codeOptions.trimRequestBody] : whether to trim request body fields
+ * @param {Boolean} [codeOptions.addCacheHeader] : whether to add cache-control header to postman SDK-request
+ * @param {Boolean} [codeOptions.followRedirect] : whether to allow redirects of a request
+ * @param {Function} callback - Callback function to return results
+ */
+function generate (source, sdkOptions, codeOptions, callback) {
```

Comment:
Source could be an object with: `{ 'type': 'string', data: collection}`. 
Lets just stick to two types of input here, string or json for now.

Reply:
Okay will shift to this approach